tests/sgl/core/test_thread.cpp

```
@@ -96,6 +140,62 @@ TEST_CASE("rhi task pool executes tasks and deletes payloads")
+        {
+            pool->waitAndReleaseTask(task);
+            std::lock_guard lock(payload.mutex);
+            payload.wait_returned = true;
+            payload.condition.notify_all();
+        }
+    );
+
+    {
+        std::unique_lock lock(payload.mutex);
+        payload.allow_callback_return = true;
+        payload.condition.notify_all();
```

Cppcheck (2.21.0)

[style] The function 'parallel_for' is never used.

(unusedFunction)